### ISSUE TITLE
In list env, contents of hboxed "tcblisting" moved to right

### ISSUE THREAD

**Issue by muzimuzhi** [T-F-S/tcolorbox#113]
From TeX-SX question https://tex.stackexchange.com/q/563252.

When used inside list environments like `itemize`, contents of `tcblisting` with options `capture=hbox, listing only` is wrongly moved to right.

An example:
```tex
\documentclass{article}
\usepackage[listings,skins]{tcolorbox}

\tcbset{listing only}

\begin{document}
\begin{itemize}
    \item \texttt{capture=minipage}
\begin{tcblisting}{}
blah
\end{tcblisting}

    \item \texttt{capture=hbox}
\begin{tcblisting}{hbox}
blah
\end{tcblisting}

    \begin{itemize}
        \item \texttt{capture=minipage}, second list level
\begin{tcblisting}{}
blah
\end{tcblisting}

        \item \texttt{capture=hbox}, second list level
\begin{tcblisting}{hbox}
blah
\end{tcblisting}
    \end{itemize}
\end{itemize}

outside list
\begin{tcblisting}{}
blah
\end{tcblisting}

\begin{tcblisting}{hbox}
blah
\end{tcblisting}
\end{document}
```

![image](https://user-images.githubusercontent.com/6376638/93640265-dc50a880-fa2c-11ea-872e-3e8197ffb409.png)

**Comment by T-F-S** [comment]
The shift problem is caused by `\@totalleftmargin` which is increased inside an `itemize` environment. In a *normal* tcolorbox, the content is put inside a `minipage` where `\@totalleftmargin` is set to zero. For a `hbox` type tcolorbox, there is currently no such setting. It behaves like `\fbox`.

But, I think it would be useful to add some `minipage` features to the `hbox` type to solve such problems in a general way. The following patch adds `\@parboxrestore` from `minipage` to `hbox`:

```latex
\long\def\tcbox@inner@hbox#1{%
  \tcbset{breakable@false,sidebyside=false}%
  \tcb@set@@phantom%
  \sbox\tcb@upperbox{\tcb@embed@tcbox{%
    \csname tcb@parbox@use@\kvtcb@parbox\endcsname%
    \@parboxrestore%
    \color{tcbcolupper}\kvtcb@fontupper\kvtcb@halignupper\tcb@insert@before@upper#1\tcb@insert@after@upper}}%
  \tcbdimto\tcb@val@raisebase{\the\dimexpr\dp\tcb@upperbox+\kvtcb@bottom+\kvtcb@boxsep+\kvtcb@bottom@rule@stand+\kvtcb@bbbottom@stand\relax}%
  \tcbdimto\kvtcb@width{\wd\tcb@upperbox+\kvtcb@left@rule+\kvtcb@leftupper+\kvtcb@boxsep*2+\kvtcb@rightupper+\kvtcb@right@rule}%
  \tcb@hasLowerfalse%
  \tcb@set@@title%
  \tcb@set@@dimensions%
  \tcb@draw@color@box%
}

```

**Comment by MonsieurPi** [comment]
Hey, thanks for the fix, how should I use it in my file? I tried putting it just below the `\tcbset{listing only}` but nothing changed.

And actually I got another problem and I think it's related but your fix makes the compilation fail:

```latex
\documentclass{article}

\usepackage{listings}
\usepackage[]{tcolorbox}

\newcommand{\test}[1]{\tcbox[]{\lstinline|#1|}}
\newcommand{\testgood}[1]{\tcbox[]{#1}}

\begin{document}

\begin{minipage}[t]{0.5\linewidth}
  \begin{itemize}
    \item asd
    \item \test{blah}
    \begin{itemize}
      \item asd
      \item \test{blah}
    \end{itemize}
  \end{itemize}
\end{minipage}
\begin{minipage}[t]{0.5\linewidth}
  \begin{itemize}
    \item asd
    \item \testgood{blah}
    \begin{itemize}
      \item asd
      \item \testgood{blah}
    \end{itemize}
  \end{itemize}
\end{minipage}
\end{document}
```

Outputs:

![image](https://user-images.githubusercontent.com/35367400/93753787-fe028900-fc00-11ea-9529-a243992b3eff.png)

(looks like the `lstinline` is making `\tbox` bug but there is no `\tcblistinginline` so I didn't know how to do it differently ;-))

But when I put your fix before `\begin{document}`

>! Use of \tcbox doesn't match its definition.
\test #1->\tcbox [
                  ]{\lstinline |#1|}
l.28   \item \test{blah}

**Comment by T-F-S** [comment]
Implemented in version 4.40 (2020/09/25).
